pandas/_libs/tslibs/parsing.pyx

```diff
@@ -421,6 +421,10 @@ cdef parse_datetime_string_with_reso(
             parsed = datetime(
                 dts.year, dts.month, dts.day, dts.hour, dts.min, dts.sec, dts.us
             )
+            if out_bestunit == NPY_DATETIMEUNIT.NPY_FR_ns:
+                # No picoseconds, so no nanoseconds.
+                # Have to have seen microseconds, in order to have "seen" nanoseconds
+                out_bestunit = NPY_DATETIMEUNIT.NPY_FR_us
```

Comment:
i think doing this will mess up Timestamp inference in a follow-up to #49737

Reply:
Yeah, I only added this block because somehow CI was failing tests only on Linux(maybe a locale problem). 
(Issue was that datetime object was returned, when reso was nanosecond, and datetimes don't have nanosecond support).

https://github.com/pandas-dev/pandas/blob/7ef6a71c5c41eadf9ebea514f4be502e83c95a6f/pandas/_libs/tslibs/parsing.pyx#L413 seems buggy.

IMO, it should probably be possible to have nanoseconds even when picoseconds is zero, so we should be returning timestamp as long as ``out_bestunit == NPY_DATETIMEUNIT.NPY_FR_ns``. I don't know if that will break anything, though.

Reply:
Ok, I've updated this block now to checking the bestunit.

@jbrockmendel @WillAyd Would this work fine?